vizro-core/src/vizro/models/_models_utils.py

```diff
@@ -16,7 +18,27 @@ def _wrapper(self, *args, **kwargs):
 
 
 # Validators for reuse
-def _validate_min_length(cls, field):
+def validate_min_length(cls, field):
     if not field:
         raise ValueError("Ensure this value has at least 1 item.")
     return field
+
+
+def check_captured_callable(cls, field):
+    mode_to_error = {
+        "figure": "A callable of mode `figure` has been provided. Please wrap it inside the `vm.Figure(figure=...)`.",
+        "table": "A callable of mode `table` has been provided. Please wrap it inside the `vm.Table(figure=...)`.",
+        "ag_grid": "A callable of mode `ag_grid` has been provided. Please wrap it inside the `vm.AgGrid(figure=...)`.",
+        "graph": "A callable of mode `graph` has been provided. Please wrap it inside the `vm.Graph(figure=...)`.",
+    }
+
+    for component in field:
+        if isinstance(component, (_DashboardReadyFigure, CapturedCallable)):
+            mode = (
+                component._captured_callable._mode if isinstance(component, _DashboardReadyFigure) else component._mode
+            )
+            error_message = mode_to_error.get(mode)
+            if error_message:
+                raise ValueError(error_message)
+
+    return field
```

Comment:
@antonymilne - I was initially very confused why my unit tests passed for `dash_ag_grid, dash_data_table and kpi_card` but not for any `px chart` until I saw that the type of the px.chart is not `CapturedCallable` as for the others but it's _DashboardReadyFigure. Similar to access the mode, it's different for these. 

Is there a reason for this? Would you do the check as I have it or is there a better way to check the mode etc?

Reply:
Hmmm yes, this was a bit more fiddly than I had anticipated. You did a great job of it - and I have just rewritten it a bit - see these commits:
![image](https://github.com/user-attachments/assets/a67c3b59-7a41-4763-bb91-007b80bba986)

Reply:
I like it! 🚀